flytekit/core/type_engine.py

```diff
@@ -2333,10 +2333,8 @@ def generate_attribute_list_from_dataclass_json(schema: dict, schema_name: typin
                 )
             elif property_val.get("additionalProperties"):
                 attribute_list.append(
-                    (property_key, typing.cast(GenericAlias, _get_element_type(property_val["additionalProperties"]))),
+                    (property_key, Dict[str, _get_element_type(property_val["additionalProperties"])])  # type: ignore[misc,index]
                 )
-            else:
-                attribute_list.append((property_key, Dict[str, _get_element_type(property_val)]))  # type: ignore[misc,index]
         # Handle int, float, bool or str
```

Comment:
Can you explain more why they can be merged together to `Dict[str, _get_element_type(property_val["additionalProperties"])])`?

Reply:
Both `dict` and `typing.Dict` include the `additionalProperties` field. The key difference is that the `additionalProperties` value from `dict` is an empty dictionary, whereas the one from `typing.Dict` contains type information. I plan to make a change to address this distinction.